PROJECT_GUIDELINES.md

```diff
@@ -0,0 +1,173 @@
+
+##### i18n
+
+Components should require as a few text translations as possible. In general lets users supply text values via slots and attributes. The lets user handle translations with their apps.
+
+If you component involves formatting numbers or dates use the [`Intl` APIs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) for formating the display of numbers and dates in your component.
+
+To add RTL support to your components you should use the internal `getElementDir` helper to apply the `dir` attribute to your component. This means that your components `dir` attribute will always match the documents `dir`.
+
+```tsx
+import { Component, Host, Element, h} from "@stencil/core";
+import { getElementDir } from "../../utils/dom";
+
+@Component({
+  tag: "calcite-component",
+  styleUrl: "calcite-component.scss",
+  shadow: true
+})
+export class CalciteComponent {
+  @Element() el: HTMLElement;
+
+  // ...
+
+  render() {
+    const dir = getElementDir(this.el);
+
+    return (
+      <Host dir={dir}>
+        <!-- The rest of your component -->
+      </Host>
+    );
+  }
+}
+```
+
+You can then implement direction specific CSS with CSS variables:
+
+```scss
+:host {
+  --calcite-tabs-tab-margin-start: 1.25rem;
+  --calcite-tabs-tab-margin-end: 0;
+}
+
+:host([dir="rtl"]) {
+  --calcite-tabs-tab-margin-start: 0;
+  --calcite-tabs-tab-margin-end: 1.25rem;
+}
+```
+
+Your component and its child components can then use `var(--calcite-tabs-tab-margin-start)` to access their proper values based on the direction of the document.
+
+In future it will likely become necessary to provide sting translations for components. An example would be the `aria-label` for the `<calcite-modal>` close button. Initial research looks promising and we could likely implement one of these approaches and set a `lang` for each component similar to how we set `dir`.
+
+- https://medium.com/stencil-tricks/implementing-internationalisation-i18n-with-stencil-5e6559554117 and https://codesandbox.io/s/43pmx55vo9
+- https://github.com/ionic-team/ionic-stencil-conference-app/issues/69
+
+#### Testing
+
+Place test files in the component folder using the `component.e2e.ts` naming convention. Write testable code and pure functions, avoid and extract side effects. Run tests locally before making any pull requests, and please, document tests with instructions if needed.
+
+#### Browser Support
+
+Our components are best used in most modern browsers, though there have been some complications with stencil loading elements polyfills in Edge. So, some APIs that are available in Chrome and Firefox aren't available in Edge. Internet Explorer is not recommended to use with our web components.
```

Comment:
I think this can be simplified to something like "We support the following browsers...". The rest can be included in a "Gotcha" section.

![gotcha](https://user-images.githubusercontent.com/197440/73765160-01f0a200-4729-11ea-9f10-4a03def5a3ef.gif)